src/modules/vtol_att_control/vtol_type.cpp

```diff
@@ -263,5 +252,153 @@ void VtolType::check_quadchute_condition()
 			}
 		}
 	}
+}
+
+bool VtolType::set_idle_mc()
+{
+	const char *dev = PWM_OUTPUT0_DEVICE_PATH;
+	int fd = px4_open(dev, 0);
+
+	if (fd < 0) {
+		PX4_WARN("can't open %s", dev);
+		return false;
+	}
+
+	unsigned servo_count;
+	int ret = px4_ioctl(fd, PWM_SERVO_GET_COUNT, (unsigned long)&servo_count);
+	unsigned pwm_value = _params->idle_pwm_mc;
+	struct pwm_output_values pwm_values;
+	memset(&pwm_values, 0, sizeof(pwm_values));
+
+	for (int i = 0; i < _params->vtol_motor_count; i++) {
+		pwm_values.values[i] = pwm_value;
+		pwm_values.channel_count++;
+	}
+
+	ret = px4_ioctl(fd, PWM_SERVO_SET_MIN_PWM, (long unsigned int)&pwm_values);
+
+	if (ret != OK) {
+		PX4_WARN("failed setting min values");
+	}
+
+	px4_close(fd);
+
+	return ret == PX4_OK;
+}
+
+bool VtolType::set_idle_fw()
```

Comment:
Why can't we do this with set_motor_state()?

Reply:
@dagar The problem is that these are two different states which need to be tracked separately.

Reply:
I mainly wanted to share as much of the code that calls that IOCTLS as we could. At a glance it looks like you've done that now.

Reply:
@dagar Yep, my last commit does that -only one method is used to do ioctl to the pwm output device.